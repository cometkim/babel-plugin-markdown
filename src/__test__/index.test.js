import fs from 'fs'
import path from 'path'
import * as babel from 'babel-core'
import plugin from '../'

test('it works', () => {
  compare({
    input: 'markdown`# hi ${1+1}`'
  })
})

test('it only transforms markdown', () => {
  compare({
    input: 'const a = markdown`# hi`; const b = md`# bye`'
  })
})

test('markdown-it options', () => {
  compare({
    input: fixture('opts.js').trim(),
    pluginOptions: {
      highlight() {
        return 'hahah'
      },
      plugins: ['task-lists']
    },
    babelOptions: {
      filename: __filename
    }
  })
})

test('external', () => {
  compare({
    input: 'markdown.require("./__fixtures__/external.md")',
    babelOptions: {
      filename: __filename
    }
  })
})

function compare({ input, pluginOptions, babelOptions }) {
  const { code } = babel.transform(input, {
    babelrc: false,
    plugins: [[plugin, pluginOptions]],
    ...babelOptions
  })
  expect(code).toMatchSnapshot()
}

function fixture(...args) {
  return fs.readFileSync(path.join(__dirname, '__fixtures__', ...args), 'utf8')
}
