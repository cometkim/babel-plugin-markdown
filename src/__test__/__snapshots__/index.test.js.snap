// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`external 1`] = `"\\"<h1>hi</h1>\\\\n\\";"`;

exports[`it only transforms markdown 1`] = `"const a = \\"<h1>hi</h1>\\\\n\\";const b = md\`# bye\`;"`;

exports[`it works 1`] = `"\\"<h1>hi 2</h1>\\\\n\\";"`;

exports[`markdown-it options 1`] = `"\\"<h1>hi</h1>\\\\n<pre><code class=\\\\\\"language-js\\\\\\">hahah</code></pre>\\\\n<ul class=\\\\\\"contains-task-list\\\\\\">\\\\n<li class=\\\\\\"task-list-item\\\\\\"><input class=\\\\\\"task-list-item-checkbox\\\\\\" disabled=\\\\\\"\\\\\\" type=\\\\\\"checkbox\\\\\\"> hi</li>\\\\n<li class=\\\\\\"task-list-item\\\\\\"><input class=\\\\\\"task-list-item-checkbox\\\\\\" checked=\\\\\\"\\\\\\" disabled=\\\\\\"\\\\\\" type=\\\\\\"checkbox\\\\\\"> there</li>\\\\n</ul>\\\\n\\";"`;
